ruby/red-arrow/test/each-raw-record/test-basic-arrays.rb

```diff
@@ -0,0 +1,524 @@
+      # 02:00:09.987654321
+      [Arrow::Time.new(unit, (60 * 60 * 2 + 9) * 1_000_000_000 + 987_654_321)],
+    ]
+    iterated_records = []
+    target = build({
+                     column: {
+                       type: :time64,
+                       unit: :nano,
+                     }
+                   },
+                   records)
+    target.each_raw_record do |record|
+      iterated_records << record
+    end
+    assert_equal(records, iterated_records)
+  end
+
+  def test_decimal128
+    records = [
+      [BigDecimal("92.92")],
+      [nil],
+      [BigDecimal("29.29")],
+    ]
+    iterated_records = []
+    target = build({
+                     column: {
+                       type: :decimal128,
+                       precision: 8,
+                       scale: 2,
+                     }
+                   },
+                   records)
+    target.each_raw_record do |record|
+      iterated_records << record
+    end
+    assert_equal(records, iterated_records)
+  end
+
+  def test_decimal256
+    records = [
+      [BigDecimal("92.92")],
+      [nil],
+      [BigDecimal("29.29")],
+    ]
+    iterated_records = []
+    target = build({
+                     column: {
+                       type: :decimal256,
+                       precision: 38,
+                       scale: 2,
+                     }
+                   },
+                   records)
+    target.each_raw_record do |record|
+      iterated_records << record
+    end
+    assert_equal(records, iterated_records)
+  end
+
+  def test_month_interval
+    records = [
+      [1],
+      [nil],
+      [12],
+    ]
+    iterated_records = []
+    target = build({column: :month_interval}, records)
+    target.each_raw_record do |record|
+      iterated_records << record
+    end
+    assert_equal(records, iterated_records)
+  end
+
+  def test_day_time_interval
+    records = [
+      [{day: 1, millisecond: 100}],
+      [nil],
+      [{day: 2, millisecond: 300}],
+    ]
+    iterated_records = []
+    target = build({column: :day_time_interval}, records)
+    target.each_raw_record do |record|
+      iterated_records << record
+    end
+    assert_equal(records, iterated_records)
+  end
+
+  def test_month_day_nano_interval
+    records = [
+      [{month: 1, day: 1, nanosecond: 100}],
+      [nil],
+      [{month: 2, day: 3, nanosecond: 400}],
+    ]
+    iterated_records = []
+    target = build({column: :month_day_nano_interval}, records)
+    target.each_raw_record do |record|
+      iterated_records << record
+    end
+    assert_equal(records, iterated_records)
+  end
+end
+
+class EachRawRecordRecordBatchBasicArraysTest< Test::Unit::TestCase
+  include EachRawRecordBasicArraysTests
+
+  def build(schema, records)
+    Arrow::RecordBatch.new(schema, records)
+  end
+end
+
+class EachRawRecordTableBasicArraysTest < Test::Unit::TestCase
+  include EachRawRecordBasicArraysTests
+
+  def build(schema, records)
+    Arrow::Table.new(schema, records)
```

Comment:
Could you create multiple chunked arrays?

```suggestion
    Arrow::Table.new(schema,
                     [
                       Arrow::RecordBatch.new(schema, records[0, 2]),
                       Arrow::RecordBatch.new(schema, records[2..-1]),
                     ])
```

Reply:
fix: [34b53e0](https://github.com/apache/arrow/pull/37137/commits/34b53e0a3ed95788cdcc617b010093636a3ad8c3)